x-pack/plugins/index_management/public/application/components/component_templates/component_template_details/component_template_details.tsx

```diff
@@ -76,21 +86,62 @@ export const ComponentTemplateDetailsFlyoutContent: React.FunctionComponent<Prop
       </SectionLoading>
     );
   } else if (error) {
-    content = (
-      <EuiCallOut
-        title={
-          <FormattedMessage
-            id="xpack.idxMgmt.componentTemplateDetails.loadingErrorMessage"
-            defaultMessage="Error loading component template"
-          />
-        }
-        color="danger"
-        iconType="warning"
-        data-test-subj="sectionError"
-      >
-        <p>{error.message}</p>
-      </EuiCallOut>
-    );
+    if (
+      error?.error === 'Not Found' &&
+      isIntegrationsComponentTemplate(decodedComponentTemplateName)
+    ) {
+      content = (
+        <EuiCallOut
+          title={
+            <FormattedMessage
+              id="xpack.idxMgmt.componentTemplateDetails.createMissingIntegrationTemplate.calloutTitle"
+              defaultMessage="Custom template doesn't exist"
+            />
+          }
+          color="warning"
+          iconType="warning"
+          data-test-subj="missingCustomComponentTemplate"
+        >
+          <p>
+            <FormattedMessage
+              id="xpack.idxMgmt.componentTemplateDetails.createMissingIntegrationTemplate.text"
+              defaultMessage="The custom template for {templateName} doesn't exist."
```

Comment:
nit: Wouldn't it be better to say `"The custom template {templateName} doesn't exist."` (without "for")? Otherwise, it sounds as if this template is not the same as the mentioned template name. The ui-copy team might provide a better suggestion. 😄